infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java

```diff
@@ -48,14 +49,16 @@
   private final Pair<String, String> projectsRootPathEnvVar;
 
   public MachineResolver(
-      String pluginId,
+      String pluginPublisher,
+      String pluginName,
       Pair<String, String> projectsRootPathEnvVar,
       Container container,
       CheContainer cheContainer,
       String defaultSidecarMemoryLimitBytes,
       List<ChePluginEndpoint> containerEndpoints,
       Map<String, String> wsAttributes) {
-    this.pluginId = pluginId;
+    this.pluginName = pluginName;
+    this.pluginPublisherName = pluginPublisher + "/" + pluginName;
```

Comment:
Could we use `pluginPublisher` (without `Name`)? It reads slightly better and the `Name` part seems redundant.

Reply:
It is actually publisher + name. Renamed to `pluginPublisherAndName`